### PR TITLE
Fix unit tests wrt. latest dev/IPv6_Integration changes

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR fixes the unit test failures for all unit tests modules which were failing after the latest IPv6 integration source changes. The `xEndPoint` structures that are used in the test cases are now initialized to zero, to prevent uninitialized data from previous tests affecting the behavior of the tests.

The CBMC proof for `DNSgetHostByName_a` is updated to not assert on `pxAddressInfo != NULL` during the DNS callback as `pxAddressInfo` being NULL is a valid case during timeouts. 

Test Steps
-----------
```
cmake -S test/unit-test -B test/unit-test/build/ 
make -C test/unit-test/build/ all 
cd test/unit-test/build/
ctest -E system --output-on-failure --verbose
```

Checklist:
----------
- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
Unit test cases were failing due to the changes in the source code as part of the IPv6 integration.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/FreeRTOS_DHCP/FreeRTOS_DHCP_utest.c [hunk 1]
```diff
@@ -507,51 +507,51 @@ void test_vDHCPProcess_ResetAndIncorrectStateWithRNGSuccessSocketCreationFail( v
...
-        /* Ignore the inputs to setting the socket options. */
-        FreeRTOS_setsockopt_ExpectAnyArgsAndReturn( pdPASS );
-        FreeRTOS_setsockopt_ExpectAnyArgsAndReturn( pdPASS );
-        /* Make sure that binding fails. Return anything except zero. */
-        vSocketBind_ExpectAnyArgsAndReturn( pdTRUE );
-
-        catch_assert( vDHCPProcess( pdTRUE, pxEndPoint ) );
-    }
-}
+// void test_vDHCPProcess_ResetAndIncorrectStateWithRNGSuccessSocketBindFail( void )
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_DHCP/FreeRTOS_DHCP_utest.c:510]: Can we remove the test?

File: test/cbmc/proofs/DNS/DNSgetHostByName_a/DNSgetHostByName_a_harness.c [hunk 2]
```diff
@@ -95,8 +95,8 @@ void func( const char * pcHostName,
                       "Precondition: pcHostName != NULL" );
     __CPROVER_assert( pvSearchID != NULL,
                       "Precondition: pvSearchID != NULL" );
-    __CPROVER_assert( pxAddressInfo != NULL,
-                      "Precondition: pxAddressInfo != NULL" );
+    /* pxAddressInfo is not validated for not being NULL as
```
Comment (moninom1) [inline test/cbmc/proofs/DNS/DNSgetHostByName_a/DNSgetHostByName_a_harness.c:98]: Is this something new?

**Review by moninom1 — APPROVED** [review]

File: test/unit-test/FreeRTOS_DHCP/FreeRTOS_DHCP_utest.c (hunk 1 above)
Comment (tony-josi-aws) [reply test/unit-test/FreeRTOS_DHCP/FreeRTOS_DHCP_utest.c]: Uncommented

**Review by shubnil — APPROVED** [review]

**Pull request merged**